### PR TITLE
add test for capacity pools

### PR DESCRIPTION
**Pull request by mhost39**
### Related Issues
https://github.com/threefoldtech/js-sdk/issues/1722

### CODE REVIEW

File: tests/frontend/pages/pools/pools.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        self.click_button("CLOSE")
+
+    def wait(self, class_name):
+        wait = WebDriverWait(self.driver, 60)
+        wait.until(EC.invisibility_of_element_located((By.CLASS_NAME, class_name)))
+
+    def click_button(self, text):
+        buttons = self.driver.find_elements_by_class_name("v-btn")
+        next_button = [button for button in buttons if button.text == text][0]
+        next_button.click()
```
Comment (AhmedHanafy725) [inline tests/frontend/pages/pools/pools.py:47]: lets put these helper methods in the [base](https://github.com/threefoldtech/js-sdk/blob/development/tests/frontend/pages/base.py) to be used in all pages

File: tests/frontend/pages/pools/pools_solution.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        self.click_button("FINISH")
+
+    def click_button(self, text):
+        buttons = self.driver.find_elements_by_class_name("v-btn")
+        next_button = [button for button in buttons if button.text == text][0]
+        next_button.click()
+
+    def wait(self, class_name):
+        wait = WebDriverWait(self.driver, 60)
+        wait.until(EC.invisibility_of_element_located((By.CLASS_NAME, class_name)))
```
Comment (AhmedHanafy725) [inline tests/frontend/pages/pools/pools_solution.py:150]: This should be used from the base after moving them.

File: tests/frontend/pages/pools/pools_solution.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        wallets = wallets_box.find_elements_by_class_name("v-radio")
+        wallet = [wallet for wallet in wallets if wallet.text == wallet_name][0]
+        wallet.click()
+
+        self.click_button("NEXT")
+        self.wait("v-progress-circular")
+        self.wait("progressbar")
+        self.click_button("FINISH")
+
+    def extend(self, wallet_name, name, cu=1, su=1, duration_unit="Day", time_to_live=1):
```
Comment (AhmedHanafy725) [inline tests/frontend/pages/pools/pools_solution.py:80]: `extend` and `create` methods are almost have the same code, can we get the common code in a helper method, then use it to create and extend?

File: tests/frontend/pages/wallets/wallets.py [hunk 4]
```diff
@@ -28,6 +28,42 @@ def create(self, name):
         wait = WebDriverWait(self.driver, 60)
         wait.until(EC.invisibility_of_element_located((By.CLASS_NAME, "v-dialog")))
 
+    def add_funded(self, name):
```
Comment (AhmedHanafy725) [inline tests/frontend/pages/wallets/wallets.py:31]: Why wallet module is here? It is already in another PR.

File: tests/frontend/tests/base_tests.py [hunk 5]
```diff
@@ -70,9 +70,9 @@ def setUp(self):
         options = Options()
         options.add_argument("--no-sandbox")
         # For browser's head mode comment the next 3 lines
-        options.add_argument("headless")
-        options.add_argument("--disable-gpu")
-        options.add_argument("--disable-dev-shm-usage")
+        # options.add_argument("headless")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/base_tests.py:73]: Please uncomment these lines to run them on CI

File: tests/frontend/tests/test_pools.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        wallet_name = self.random_name()
+        wallets = Wallets(self.driver)
+        wallets.load()
+        wallets.add_funded(wallet_name)
+
+        self.info("Create a pool")
+        pool_name = self.random_name().lower()
+        pools = PoolsSolution(self.driver)
+        pools.load()
+        pools.create(name=pool_name, wallet_name=wallet_name)
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:30]: Please create the pool with random values for cus and sus to make sure that these values reflect correctly

File: tests/frontend/tests/test_pools.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        """Test case for extending a pool.
+
+        **Test Scenario**
+        - Create a wallet.
+        - Create a pool.
+        - Extend this pool.
+        - Check that pool has been Extended.
+        - Delete the wallet.
+        """
+        self.info("Create a wallet.")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:52]: Can we create this wallet in the setUpClass to be used in all tests?

File: tests/frontend/tests/test_pools.py [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    def test01_create_pool(self):
+        """Test case for creating a pool.
+
+        **Test Scenario**
+        - Create a wallet.
+        - Create a pool.
+        - Check that the pool has been created in the pools card.
+        - Delete the a wallet.
+        """
+        self.info("Create a wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:20]: Can we create this wallet in the setUpClass to be used in all tests?

File: tests/frontend/tests/test_pools.py (hunk 6 above)
Comment (AhmedHanafy725) [reply tests/frontend/tests/test_pools.py:30]: Then we can check also the pool has been created with the right amount of cus and sus

File: tests/frontend/tests/test_pools.py [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+        pools.load()
+        pools.create(name=pool_name, wallet_name=wallet_name)
+
+        self.info("Check that the pool has been created in the pools card")
+        test_pools = Pools(self.driver)
+        test_pools.load()
+        pools_name = [name[0] for name in test_pools.list()]
+        self.assertIn(pool_name, pools_name)
+
+        self.info("Delete the wallet.")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:38]: Deleting the wallet should be in tearDownclass

File: tests/frontend/tests/test_pools.py [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        )
+
+        self.info("Check that pool has been Extended")
+        test_pools = Pools(self.driver)
+        test_pools.load()
+        calculated_su = (su + 1) * time_to_live * 60 * 60 * 24
+        calculated_cu = (cu + 1) * time_to_live * 60 * 60 * 24
+        self.assertIn((pool_name, float(calculated_cu), float(calculated_su)), test_pools.list())
+
+        self.info("Delete the wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:82]: Deleting the wallet should be in tearDownclass

File: tests/frontend/tests/test_pools.py [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+        **Test Scenario**
+        - Create a wallet.
+        - Create a pool.
+        - Check that the pool has been created in the pools card.
+        - Hide this pool.
+        - Check that the pool has been hidden from the pools card.
+        - Delete the wallet.
+        """
+        self.info("Create a wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:97]: Can we create this wallet in the setUpClass to be used in all tests?

File: tests/frontend/tests/test_pools.py [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+        self.info("Hide this pool")
+        test_pools.load()
+        test_pools.hide_pool(pool_name)
+
+        self.info("Check that the pool has been hidden from the pools card")
+        test_pools.load()
+        pools_name = [name[0] for name in test_pools.list()]
+        self.assertNotIn(pool_name, pools_name)
+
+        self.info("Delete the wallet")
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:124]: Deleting the wallet should be in tearDownclass

File: tests/frontend/tests/test_wallets.py [hunk 13]
```diff
@@ -30,3 +30,66 @@ def test_create_delete_wallet(self):
         self.info("Check that the wallet has been deleted from the wallets cards.")
         all_wallets = wallets.list()
         self.assertNotIn(wallet_name, all_wallets.keys())
+
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_wallets.py:33]: wallet tests are already in another PR.

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
Please generate the docs for these tests

File: tests/frontend/pages/wallets/wallets.py (hunk 4 above)
Comment (mhost39) [reply tests/frontend/pages/wallets/wallets.py:31]: i need to create a wallet for test

File: tests/frontend/tests/test_pools.py [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+from tests.frontend.tests.base_tests import BaseTest
+
+
+@pytest.mark.integration
+class PoolsTests(BaseTest):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.wallet_name = cls.random_name().lower()
+        j.clients.stellar.create_testnet_funded_wallet(cls.wallet_name)
```
Comment (AhmedHanafy725) [inline tests/frontend/tests/test_pools.py:15]: This is not valid anymore as we use STD wallets

File: tests/frontend/pages/pools/pools_solution.py [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+        self.set_pool_details(cu=cu, su=su, duration_unit=duration_unit, time_to_live=time_to_live)
+
+        # select the wallet
+        self.select_wallet(wallet_name)
+
+        self.wait(self.driver, "v-progress-circular")
+        self.wait(self.driver, "progressbar")
+        self.click_button(self.driver, "FINISH")
+
+    def choose_option(self, option):
```
Comment (AhmedHanafy725) [inline tests/frontend/pages/pools/pools_solution.py:81]: can we use a better name for this method?

Comment (AhmedHanafy725) [reply tests/frontend/pages/pools/pools_solution.py:81]: Also, this needs to add a docstring to know what is the options to be chosen

File: tests/frontend/pages/pools/pools_solution.py [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+        name_input.send_keys(name)
+        self.click_button(self.driver, "NEXT")
+
+        # set cu, su and time to live to input elements
+        self.set_pool_details(cu=cu, su=su, duration_unit=duration_unit, time_to_live=time_to_live)
+
+        # select the Freefarm
+        farm_box = self.driver.find_element_by_class_name("v-select__selections")
+        farm_input = farm_box.find_element_by_tag_name("input")
+        farm_input.send_keys("Freefarm")
```
Comment (AhmedHanafy725) [inline tests/frontend/pages/pools/pools_solution.py:37]: The farm shouldn't be hard coded, it should selected for the test.

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]
all helper methods in the pools module should be hidden

**Review by AhmedHanafy725 — APPROVED** [review]

**Review by waleedhammam — APPROVED** [review]

**Pull request merged**